public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -216,8 +226,18 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     const payload = angular.toJson(header) + '\n' + angular.toJson(data) + '\n';
 
     return this.post('_msearch', payload).then((res: any) => {
+      let parsedRes = undefined;
+      if (typeof res !== 'object') {
+        try {
+          parsedRes = JSON.parse(res);
+        } catch (err) {
+          throw err;
```

Comment:
Is there a reason for catching and immediately throwing the error? Seems like we could just not have a try catch here.